packages/database/src/migrations/20210420031825-WeeklyCasesTrendGraphForPalauCountry-modifies-data.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+import { insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const DISEASE_DATA_ELEMENTS = {
+  AFR: {
+    dataElement: 'PSSS_AFR_Total_Cases',
+    name: 'Acute Fever and Rash',
+    color: '#F0965BFF', // orange
+  },
+  DIA: {
+    dataElement: 'PSSS_DIA_Total_Cases',
+    name: 'Diarrhoea',
+    color: '#81DEE4FF', // aqua
+  },
+  ILI: {
+    dataElement: 'PSSS_ILI_Total_Cases',
+    name: 'Influenza Like Illness',
+    color: '#4DA347FF', // green
+  },
+  PF: {
+    dataElement: 'PSSS_PF_Total_Cases',
+    name: 'Prolonged Fever',
+    color: '#1C49A7FF', // blue
+  },
+  DLI: {
+    dataElement: 'PSSS_DLI_Total_Cases',
+    name: 'Dengue Like Illness',
+    color: '#8455F6', // purple
+  },
+  CON: {
+    dataElement: 'PSSS_CON_Total_Cases',
+    name: 'Conjunctivitis',
+    color: '#BE72E0', // pink
+  },
+};
+
+const getDashboardReportId = diseaseCode =>
+  `PSSS_PW_${diseaseCode}_Weekly_Case_Trend_Graph_Country`;
+
+const dashboardGroupCode = 'PW_PSSS_Syndromic_Surveillance_National_Data_Country_Public';
+
+const getDashboardReport = (id, diseaseName, dataElementCode, color) => ({
+  id,
+  dataBuilder: 'sumPerPeriod',
+  dataBuilderConfig: {
+    dataClasses: {
+      value: {
+        codes: [dataElementCode],
+      },
+    },
+    aggregationType: 'FILL_EMPTY_WEEK_WITH_NULL',
+    entityAggregation: {
+      aggregationOrder: 'BEFORE',
+      dataSourceEntityType: 'facility',
+      aggregationEntityType: 'country',
+    },
+  },
+  viewJson: {
+    name: `${diseaseName} Weekly Case Number Trend Graph`,
+    type: 'chart',
+    chartType: 'line',
+    chartConfig: {
+      value: {
+        color,
+      },
+    },
+    defaultTimePeriod: {
+      end: {
+        unit: 'week',
+        offset: 0,
+      },
+      start: {
+        unit: 'week',
+        offset: -52,
+      },
+    },
+    presentationOptions: {
+      periodTickFormat: '[W]w',
```

Comment:
Ahh I've been wondering how to do this, looks awesome will try it out!

Reply:
I thought that was a new functionality as well.